plugin.video.hbogoeu/hbogolib/handlereu.py

```diff
@@ -0,0 +1,1447 @@
+                liz.setProperty("resumetime", str(0))
+                liz.setInfo(type="Video", infoLabels={"PlayCount": "1"})
+                self.log(cid + " External ID: " + externalid + " IS WATCHED")
+        if title['ContentType'] == 1:
+            media_id = cid
+            try:
+                media_id = title['Id']
+            except KeyError:
+                pass  # all is ok got from first method just ignore
+            except Exception:
+                self.log("Unexpected error for get media id: " + traceback.format_exc())
+            liz.addContextMenuItems(items=self.genContextMenu(cid, media_id, HbogoConstants.CONTEXT_MODE_MOVIE))
+        if title['ContentType'] == 3:
+            liz.addContextMenuItems(items=self.genContextMenu(cid, cid, HbogoConstants.CONTEXT_MODE_EPISODE))
+        xbmcplugin.addDirectoryItem(handle=self.handle, url=item_url, listitem=liz, isFolder=False)
+
+    def addDir(self, item, mode, media_type):
+        if self.lograwdata:
+            self.log("Adding Dir: " + str(item) + " MODE: " + str(mode))
+        directory_url = "%s?%s" % (self.base_url, urlencode({
+            'url': item['ObjectUrl'],
+            'mode': mode,
+            'name': '%s (%d)' % (py2_encode(item['OriginalName']), item['ProductionYear'])
+        }))
+        liz = xbmcgui.ListItem(item['Name'])
+        liz.setArt({
+            'thumb': item['BackgroundUrl'], 'poster': item['BackgroundUrl'], 'banner': item['BackgroundUrl'],
+            'fanart': item['BackgroundUrl']
+        })
+        plot = py2_encode(item['Abstract'])
+        if 'Description' in item:
+            if item['Description'] is not None:
+                plot = py2_encode(item['Description'])
+        liz.setInfo(type="Video", infoLabels={
+            "mediatype": media_type, "season": item['Tracking']['SeasonNumber'],
+            "tvshowtitle": item['Tracking']['ShowName'],
+            "title": item['Name'],
+            "Plot": plot
+        })
+        liz.setProperty('isPlayable', "false")
+        if media_type == "tvshow":
+            cid = item['ObjectUrl'].rsplit('/', 2)[1]
+            media_id = cid
+            try:
+                media_id = item['SeriesId']
+            except KeyError:
+                pass  # all is ok media id got from first method ignore
+            except Exception:
+                self.log("Unexpected get media id error: " + traceback.format_exc())
+            liz.addContextMenuItems(items=self.genContextMenu(cid, media_id, HbogoConstants.CONTEXT_MODE_DEFAULT))
+        xbmcplugin.addDirectoryItem(handle=self.handle, url=directory_url, listitem=liz, isFolder=True)
+
+    def addCat(self, name, url, icon, mode):
+        if self.lograwdata:
+            self.log("Adding Cat: " + str(name) + "," + str(url) + "," + str(icon) + " MODE: " + str(mode))
+        category_url = '%s?%s' % (self.base_url, urlencode({
```

Comment:
bad practice to use `file` to name a var

Reply:
Fixed tnx